flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredOperatorBackendStateMetaInfo.java

```diff
@@ -175,14 +176,19 @@ public boolean equals(Object obj) {
 				return false;
 			}
 
+			if (!(obj instanceof Snapshot)) {
+				return false;
+			}
+
+			Snapshot snapshot = (Snapshot)obj;
+
 			// need to check for nulls because serializer and config snapshots may be null on restore
-			return (obj instanceof Snapshot)
-				&& name.equals(((Snapshot) obj).getName())
-				&& assignmentMode.equals(((Snapshot) obj).getAssignmentMode())
-				&& ((partitionStateSerializer == null && ((Snapshot) obj).getPartitionStateSerializer() == null)
-					|| partitionStateSerializer.equals(((Snapshot) obj).getPartitionStateSerializer()))
-				&& ((partitionStateSerializerConfigSnapshot == null && ((Snapshot) obj).getPartitionStateSerializerConfigSnapshot() == null)
-					|| partitionStateSerializerConfigSnapshot.equals(((Snapshot) obj).getPartitionStateSerializerConfigSnapshot()));
+			return Objects.equals(name, snapshot.getName())
+				&& Objects.equals(assignmentMode, snapshot.getAssignmentMode())
+				&& ((partitionStateSerializer == null && (snapshot.getPartitionStateSerializer() == null)
+					|| (Objects.equals(partitionStateSerializer, snapshot.getPartitionStateSerializer())))
```

Comment:
```
&& ((partitionStateSerializer == null && (snapshot.getPartitionStateSerializer() == null)
    || (Objects.equals(partitionStateSerializer, snapshot.getPartitionStateSerializer())))
```

These two lines can just be `&& Objects.equals(partitionStateSerializer, snapshot.getPartitionStateSerializer())`, because the `Objects.equals` method returns `true` if both arguments are `null`.